core/src/main/java/feast/core/auth/authentication/GoogleOID/GoogleOpenIDAuthenticationProvider.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.auth.authentication.GoogleOID;
+
+import java.util.Map;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider;
+
+/**
+ * Google Open ID Authentication Provider. This provider is used to validate incoming requests to
+ * Feast Core.
+ */
+public class GoogleOpenIDAuthenticationProvider implements AuthenticationProvider {
```

Comment:
This seems like it would support any openid provider.  Any reason the package and class can't be made generic?

Reply:
Actually, why even have this class at all?  Why not just set up the JwtAuthenticationProvider directly?

Reply:
> This seems like it would support any openid provider. Any reason the package and class can't be made generic?

It can.

> Actually, why even have this class at all? Why not just set up the JwtAuthenticationProvider directly?

Honestly I don't recall why this was broken out. It looks like if we generalize cert handling then we can use JwtAuthenticationProvider